waspc/waspls/src/Wasp/LSP/DynamicHandlers.hs

```diff
@@ -119,3 +118,61 @@ sourceFilesChangedHandler msg = do
           logM $ "[watchSourceFilesHandler] Updating missing diagnostics for " ++ show fileUris
           publishDiagnostics uri
         Nothing -> pure ()
+
+-- | Register file watcher watcher for Prisma schema file.
+registerPrismaSchemaFileWatcher :: ServerM ()
+registerPrismaSchemaFileWatcher = do
+  -- We have two options here:
+  -- 1) Watch any schema.prisma file in the project directory tree
+  -- 2) Watch only the schema.prisma file in workspace root which might not be correct
+  --   if the user openned their IDE some other folder other than the workspace root.
+  let prismaSchemaGlob = "**/schema.prisma"
+
+  globPattern <-
+    -- NOTE: We use the workspace root, instead of the wasp root here, because
+    -- 1) The wasp root may not be known yet.
+    -- 2) Using the workspace root results only in the potential to consider
+    --    files that do not matter. Important files won't be missed (the workspace
+    --    root will necessarily contain the wasp root).
+    LSP.getRootPath >>= \case
+      Nothing -> do
+        logM "Could not access projectRootDir when setting up source file watcher. Watching any schema.prisma file instead of limiting to src/."
+        return prismaSchemaGlob
+      Just projectRootDir -> do
+        return $
+          if "/" `isSuffixOf` projectRootDir
+            then projectRootDir <> prismaSchemaGlob
+            else projectRootDir <> "/" <> prismaSchemaGlob
+
+  watchPrismaSchemaToken <-
+    LSP.registerCapability
+      LSP.SWorkspaceDidChangeWatchedFiles
+      LSP.DidChangeWatchedFilesRegistrationOptions
+        { _watchers =
+            LSP.List
+              [ LSP.FileSystemWatcher
+                  { _globPattern = Text.pack globPattern,
+                    _kind = Nothing
+                  }
+              ]
+        }
+      prismaSchemaFileChangeHandler
+  case watchPrismaSchemaToken of
+    Nothing -> logM "[registerPrismaSchemaFileWatcher] Client did not accept WorkspaceDidChangeWatchedFiles registration"
+    Just _ -> logM $ "[registerPrismaSchemaFileWatcher] WorkspaceDidChangeWatchedFiles registered for Prisma schema file. Glob pattern: " <> globPattern
+  modify (State.regTokens . State.watchPrismaSchemaToken .~ watchPrismaSchemaToken)
+
+-- | Ran when Prisma schema file changes.
+prismaSchemaFileChangeHandler :: LSP.Handler ServerM 'LSP.WorkspaceDidChangeWatchedFiles
+prismaSchemaFileChangeHandler msg = do
+  let (LSP.List uris) = fmap (^. LSP.uri) $ msg ^. LSP.params . LSP.changes
+  logM $ "[prismaSchemaFileChangeHandler] Received file changes: " ++ show uris
+  sendToReactor $ do
+    maybeUri <- handler $ asks (^. State.waspFileUri)
+
+    case maybeUri of
+      Just uri -> do
+        logM "[prismaSchemaFileChangeHandler] Running Wasp diagnostics after change in schema.prisma"
+        -- We run the Wasp file diagnostics since those run the Prisma diagnostics as well.
```

Comment:
Our set up for Prisma file support in the Wasp LS is roughly:

1. We watch Prisma files for change in the workspace (we sometimes watch more Prisma files if the workspace contains multiple of them)
2. On change of the Prisma files: rerun _Wasp file analysis_. This in turn find the Prisma file next to the Wasp file and parses it.
3. We have handlers for when Wasp files are opened/saved/changed: we rerun Wasp analysis which then again parses the Prisma file.

Why not run Prisma analysis directly here? Since we don't have anything smart to do with the analysis result other then save it in the state. We'll want to rerun Wasp analysis anyways since the Prisma analysis result affects it directly. 

The JS/TS code is a bit different because it publishes LS diagnostics related to the Wasp file directly. It looks at the list of ext imports, checks for errors and then publishes LS diagnostics which are visible in the Wasp file. The Prisma analysis doesn't produce such side-effects, so no need to publish anything.

I hope this clears up the intention here, I'm struggling myself with the LS code, but this is what I understood, it should be good enough IMHO.